macOS/DuckDuckGo/History/View/Onboarding/HistoryViewOnboardingPopover.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  HistoryViewOnboardingPopover.swift
+//
+//  Copyright © 2025 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import Persistence
+import SwiftUIExtensions
+
+final class HistoryViewOnboardingPopover: NSPopover {
+    let ctaCallback: (Bool) -> Void
+
+    init(ctaCallback: @escaping (Bool) -> Void) {
+        self.ctaCallback = ctaCallback
+        super.init()
+        self.behavior = .semitransient
+        setupContentController()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("\(Self.self): Bad initializer")
+    }
+
+    private func setupContentController() {
+        contentViewController = HistoryViewOnboardingViewController(ctaCallback: ctaCallback)
+    }
+}
+
+final class HistoryViewOnboardingViewController: NSHostingController<HistoryViewOnboardingView> {
+
+    init(ctaCallback: @escaping (Bool) -> Void) {
+        self.viewModel = HistoryViewOnboardingViewModel(ctaCallback: ctaCallback)
+        let view = HistoryViewOnboardingView(model: viewModel)
+        super.init(rootView: view)
+    }
+
+    @MainActor @preconcurrency required dynamic init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private let viewModel: HistoryViewOnboardingViewModel
```

Comment:
❓ I saw this in the previous file. What’s the reason for moving the constants, and variables definition after the class initialization? 

For example, the popover has it at the top; is it because it is internal?

It doesn’t seem to be consistent with the current codebase.

Reply:
Yes, it's on the bottom because it's private. We don't have rules defined for that but there is plenty of files following this convention.